src/test/java/dev/jbang/cli/TestRun.java

```diff
@@ -371,11 +370,13 @@ void testHelloWorldGAVWithNoMain() throws IOException {
 
 		assertThat(code.getResourceRef().getFile().toString(), matchesPattern(".*\\.m2.*codegen-4.5.0.jar"));
 
-		ExitException e = Assertions.assertThrows(ExitException.class,
-				() -> code.cmdGenerator(ctx).generate());
+		String result = code.cmdGenerator(ctx).generate();
+		assertThat(result, matchesPattern("^.*java(.exe)?.*"));
+		assertThat(ctx.getMainClassOr(code), nullValue());
 
-		assertThat(e.getMessage(), startsWith("no main class"));
+		assertThat(code.isJar(), equalTo(true));
```

Comment:
This can't be correct, because it would mean that trying to run a script that has no main will not result in any error for the user.

I can understand not getting an error for the `build` command, but the same can't be true for `run`.

Reply:
correct - i've restored this and then build/export still works afaics

Reply:
@quintesse @maxandersen , I would like to follow up on this detail, in the example I'm building up it's perfectly legit to `run` in interactive mode without a `main` class.

Ideally, those 2 `jbang` invocations should be only one:
https://github.com/andreaTP/playing-with-jbang/blob/758f87d9a20471e5caeaafacd88e060138a3f3f7/k8s-shell.sh#L25-L35

Would you agree to make this Main mandatory only under the conditions:
 - it's a `run` command
 - it's not interactive

Reply:
@andreaTP good point, I agree

Reply:
Yes. Agree.